portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java

```diff
@@ -1328,9 +1328,18 @@ ServletContext.class, getServletContext(),
 	private static final Snapshot<InactiveRequestHandler>
 		_inactiveRequestHandlerSnapshot = new Snapshot<>(
 			MainServlet.class, InactiveRequestHandler.class);
-	private static volatile ReleaseManager _releaseManager =
-		ServiceProxyFactory.newServiceTrackedInstance(
-			ReleaseManager.class, MainServlet.class, "_releaseManager", false);
+	private static final ReleaseManager _releaseManager;
+
+	static {
+		ServiceTracker<ReleaseManager, ReleaseManager> serviceTracker =
+			new ServiceTracker<>(
+				SystemBundleUtil.getBundleContext(), ReleaseManager.class,
+				null);
+
+		serviceTracker.open();
+
+		_releaseManager = serviceTracker.getService();
```

Comment:
@LisztLiszt This is more than what I commented. Actually, please keep getting the service when it's used. Only keep the service tracker a static field, open it in static block.